saleor/graphql/payment/mutations.py

```diff
@@ -75,6 +75,18 @@ def calculate_total(cls, info, checkout):
         )
         return max(checkout_total, zero_taxed_money(checkout_total.currency))
 
+    @classmethod
+    def clean_shipping_method(cls, checkout):
```

Comment:
Maby we should implement shipping validation rules from `clean_checkout` in `CheckoutCompleate` mutation.
[here](https://github.com/mirumee/saleor/blob/master/saleor/checkout/utils.py#L816)

We can move this function to graphql module and split it to `clean_checkout_shippint` and `clean_checkout_payment`

Reply:
For me the current solution looks fine. I'm only wondering if we have a benchmark for this mutation? `checkout.is_shipping_required()` can potentially perform a lot of DB queries.

Reply:
But this solution don't resolve all potential problems. We should use the same shipping validation in `PaymentCreate` and `CheckoutCompleate`. If you have a checkout with invalid shipping we shouldn't create the payment for this checkout.

Reply:
We have a benchmark for `checkoutPaymentCreate`(`test_checkout_payment_charge`)